mscore/pagesettings.cpp

```diff
@@ -270,310 +325,392 @@ void PageSettings::orientationClicked()
-            updateValues();
-            updatePreview(0);
-            }
+void PageSettings::etmChanged(double val)
+      {
+      QPageLayout* layout = preview->score()->style().pageEven();
+      if (!layout->setTopMargin(val)) // only error is out of range
+            layout->setTopMargin(marginMinMax(val,
+                                              layout->maximumMargins().top(), 
+                                              evenPageTopMargin));
+      updatePreview();
       }
 
 //---------------------------------------------------------
-//   otmChanged
+//   ebmChanged - even bottom margin
 //---------------------------------------------------------
 
-void PageSettings::otmChanged(double val)
+void PageSettings::ebmChanged(double val)
       {
-      if (mmUnit)
-            val /= INCH;
-      preview->score()->style().set(Sid::pageOddTopMargin, val);
-      updatePreview(1);
+      QPageLayout* layout = preview->score()->style().pageEven();
+      if (!layout->setBottomMargin(val)) // only error is out of range
+            layout->setTopMargin(marginMinMax(val,
+                                              layout->maximumMargins().bottom(),
+                                              evenPageBottomMargin));
+      updatePreview();
       }
 
+//// Left and right margins synchronize across odd/even pages
 //---------------------------------------------------------
-//   olmChanged
+//   lrMargins - helper for the left/right margin valueChanged signals
 //---------------------------------------------------------
-
-void PageSettings::olmChanged(double val)
+void PageSettings::lrMargins(double val, bool isLeft, bool isOdd, QDoubleSpinBox* spinOne)
       {
-      if (mmUnit)
-            val /= INCH;
-
-      if (twosided->isChecked()) {
-            evenPageRightMargin->blockSignals(true);
-            evenPageRightMargin->setValue(val * (mmUnit ? INCH : 1.0));
-            evenPageRightMargin->blockSignals(false);
+      MStyle&         style = preview->score()->style();
+      QPageLayout*    one   = isOdd ? style.pageOdd()  : style.pageEven();
+      QPageLayout*    other = isOdd ? style.pageEven() : style.pageOdd();
+      QDoubleSpinBox* spinOther;
+
+      bool b = isLeft ? one->setLeftMargin(val) : one->setRightMargin(val);
+      if (!b) {                    // val exceeds the max
+            double marg = isLeft ? one->maximumMargins().left()
+                                 : one->maximumMargins().right();
+            val = marginMinMax(val, marg, spinOne);
+            if (isLeft)
+                  one->setLeftMargin(val);
+            else
+                  one->setRightMargin(val);
             }
+
+      if (isLeft) {
+            other->setRightMargin(val);
+            if (twosided->isChecked())
+                  spinOther = isOdd ? evenPageRightMargin : oddPageRightMargin;
+            else 
+                  spinOther = evenPageLeftMargin; // even left margin is disabled
+      }
```

Comment:
This brace should be indented one more level.

Reply:
check